app/plugin_api.go

```diff
@@ -415,6 +415,44 @@ func (api *PluginAPI) SearchPostsInTeam(teamId string, paramsList []*model.Searc
 	return postList.ToSlice(), nil
 }
 
+func (api *PluginAPI) SearchPostsInTeamForUser(teamId string, userId string, searchParams *model.SearchParameter) (*model.PostSearchResults, *model.AppError) {
+
+	if searchParams == nil {
```

Comment:
If `nil` is an invalid value for `searchParams`, what value does a pointer provide?

Reply:
Not sure I follow the question, `searchParams` must have a pointer value to be able to perform the search.

Reply:
If `searchParams` must not be `nil`, what is the value of using a pointer vs. using a struct directly?

Reply:
Not sure, probably no extra value. 

I set it as a pointer to be consistent with how the api currently handles structs as pointer parameters in the majority of the cases.

Reply:
More recent methods like `KVSetWithOptions` did more away from using pointer all the time. I see here an opportunity to continue to break with a pattern that has not proven valuable.

Reply:
Done!